### PR TITLE
fix: allow redis url to be passed in as separate components

### PR DESCRIPTION
**Pull request by rossmcewan**
---

**Enhanced Redis Configuration: Support for URL Component-Based Setup**

This PR refactors Redis connection setup to allow the Redis URL to be constructed from separate environment variables (`NANGO_REDIS_HOST`, `NANGO_REDIS_PORT`, and `NANGO_REDIS_AUTH`) in addition to the existing `NANGO_REDIS_URL` option. The logic for building and validating the Redis connection was updated across configuration utilities and the key-value store initialization, with comprehensive updates to the environment schema validation and a significant expansion in automated test coverage to ensure all configuration variants are reliably supported.

<details>
<summary><strong>Key Changes</strong></summary>

• Refactored the `getRedisUrl` function in `packages/shared/lib/utils/utils.ts` to construct Redis URLs from `NANGO_REDIS_HOST`, `NANGO_REDIS_PORT`, and `NANGO_REDIS_AUTH` when `NANGO_REDIS_URL` is not available.
• Updated `createKVStore` and `getKVStore` in `packages/kvstore/lib/index.ts` to properly support both full-url and component-based Redis config, maintaining fallback and backward compatibility with prior setups.
• Extended Zod environment schema validation (in `packages/utils/lib/environment/parse.ts`) to include the new component-based Redis environment variables.
• Refined `destroy` function in `packages/kvstore/lib/index.ts` to correctly clear `kvstorePromise`, improving resource cleanup.
• Significantly expanded test suites for both `getKVStore` and `getRedisUrl` across `packages/kvstore/lib/index.unit.test.ts` and `packages/shared/lib/utils/utils.unit.test.ts`, covering all fallback combinations and edge cases for Redis configuration.

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• `packages/shared/lib/utils/utils.ts`
• `packages/kvstore/lib/index.ts`
• `packages/kvstore/lib/index.unit.test.ts`
• `packages/shared/lib/utils/utils.unit.test.ts`
• `packages/utils/lib/environment/parse.ts`

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/shared/lib/utils/utils.ts [hunk 1]
```diff
@@ -54,7 +54,18 @@ export function getServerBaseUrl() {
 }
 
 export function getRedisUrl() {
-    return process.env['NANGO_REDIS_URL'] || undefined;
+    const url = process.env['NANGO_REDIS_URL'];
+    if (url) {
+        return url;
+    } else {
+        const endpoint = process.env['NANGO_REDIS_ENDPOINT'];
```
Comment (bodinsamuel) [inline packages/shared/lib/utils/utils.ts]: Would hostname be clearer and more accurate?

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/lib/utils/utils.ts (hunk 1 above)
Comment (rossmcewan) [reply packages/shared/lib/utils/utils.ts]: Do you mean because it doesn't include the port? That's fair.

Comment (rossmcewan) [reply packages/shared/lib/utils/utils.ts]: Changed it to NANGO_REDIS_HOST
